src/sched/lthread.c

```diff
@@ -733,8 +745,8 @@ int lthread_create(
     lt->attr.stack_size = stack_size;
 
     /* mmap tls image */
-    lt->itlssz = libc.tls_size;
-    if (libc.tls_size)
+    lt->itlssz = (sizeof(lthread_tcb_base) + TLS_ALIGN - 1) & -TLS_ALIGN;
+    if (lt->itlssz)
```

Comment:
This logic looks confusing now.  We're setting `lt->itlssz` to a constant and then branching on that constant being non-zero.  I think the body of this if statement is reached unconditionally, so we should just remove this line.

Reply:
Good point!